src/cloud/createConnection.js

```diff
@@ -0,0 +1,81 @@
+
+/**
```

Comment:
take a look at this function and see if it makes sense. If you know of ways to clean it up I'm all ears!